Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -779,20 +781,71 @@ namespace OpenInfraPlatform {
 					//			WR2: (SIZEOF(PartOfU) = 1) XOR(SIZEOF(PartOfV) = 1) XOR(SIZEOF(PartOfW) = 1);
 					//	END_ENTITY;
 
-					std::vector<carve::geom::vector<2>> axis;
+					std::vector<carve::geom::vector<2>> axisVector;
+					carve::math::Matrix gridPositionMatrix;
 					std::vector<carve::geom::vector<2>> segmentStartPoints;
-	
 					
 					std::shared_ptr<PlacementConverterT<IfcEntityTypesT>> placementConverter = std::make_shared<PlacementConverterT<IfcEntityTypesT>>(GeomSettings(), UnitConvert());
 
 					CurveConverterT<IfcEntityTypesT> gridConv(GeomSettings(), UnitConvert(), placementConverter);
-					gridConv.convertIfcCurve2D(gridAxis->AxisCurve, axis, segmentStartPoints);
+					gridConv.convertIfcCurve2D(gridAxis->AxisCurve, axisVector, segmentStartPoints);
 
 					if (!gridAxis->SameSense)
 						// turn around the axis
-						std::reverse(std::begin(axis), std::end(axis));
+						std::reverse(std::begin(axisVector), std::end(axisVector));
+
+					if (!gridAxis->PartOfU.empty() && gridAxis->PartOfV.empty() && gridAxis->PartOfW.empty()) {
+						for (const auto& grid : gridAxis->PartOfU)
+						{
+							gridPositionMatrix = getGridPosition(grid, alreadyApplied);
+						}
+						return applyGridPositionToAxis(gridPositionMatrix, axisVector);
+					}
+					else if (gridAxis->PartOfU.empty() && !gridAxis->PartOfV.empty() && gridAxis->PartOfW.empty()) {
+						for (const auto& grid : gridAxis->PartOfV)
+						{
+							gridPositionMatrix = getGridPosition(grid, alreadyApplied);
+						}
+						return applyGridPositionToAxis(gridPositionMatrix, axisVector);
+					}
+					else if (gridAxis->PartOfU.empty() && gridAxis->PartOfV.empty() && !gridAxis->PartOfW.empty()) {
+						for (const auto& grid : gridAxis->PartOfW)
+						{
+							gridPositionMatrix = getGridPosition(grid, alreadyApplied);
+						}
+						return applyGridPositionToAxis(gridPositionMatrix, axisVector);
+					}
+					else  {
+						oip::InconsistentModellingException(gridAxis, "IfcGridAxis can only refer to a single instance of IfcGrid.");
+					}
+				}
 
-					return axis;
+				carve::math::Matrix getGridPosition(const EXPRESSReference< typename IfcEntityTypesT::IfcGrid> grid, 
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied
+				)const throw(...)
+				{
+					if (grid->ObjectPlacement) {
+						return convertIfcObjectPlacement(grid->ObjectPlacement, alreadyApplied);
+					}
+					else {
+						throw oip::InconsistentModellingException(grid, "Impossible to define a position of the grid");

```

Comment:
The correct error would be "An IfcPositioningElement must have a placement".

Siehe WHERE rule bei der Definition von `IfcPositioningElement`.

Reply:
Resolved: f1d242b